Make ioErrCount updates go-routine safe

krisis opened the pull request.

krisis committed:
Make ioErrCount updates go-routine safe

harshavardhana commented on posix.go:
```diff
@@ -39,7 +40,7 @@ const (
 	diskPath    string
 	minFreeDisk int64
-	ioErrCount  int
+	ioErrCount  *int32
```
you can keep this as int32 and use '&s.ioErrCount'

krisis committed:
Made ioErrCount int32 instead of *int32

... and implemented StorageAPI on *posix as opposed to posix type.
This is consistent with the thumb-rule that if a value of a type is
modified as part of the interface implementation then we implement the
interface on pointer to that type.
